### PR TITLE
Support Flux 1.dev

### PR DESCRIPTION
**Pull request by bjmsong**
## Motivation

https://github.com/sgl-project/sglang-jax/issues/803

## Modifications
- Add end-to-end FLUX.1-dev (from Black Forest Labs) image generation on TPU with a 3-stage pipeline: text encoding (CLIP + T5) → diffusion (FluxTransformer2DModel, FlowMatch Euler) → VAE decode (AutoencoderKL)
- Implement 5 new JAX/Flax models: CLIPTextModel, T5EncoderModel, FluxTransformer2DModel, AutoencoderKL, FlowMatchEulerDiscreteScheduler
- Add lightweight EncoderScheduler / EncoderModelRunner for multi-encoder text processing (CLIP + T5 in one stage), with JIT precompile support
- Refactor DiagonalGaussianDistribution to support both 4D image and 5D video latents
- Refactor diffusion model runner: dict-based JIT for multi-model compatibility, add FLUX forward pass (latent packing/unpacking, dynamic sigma shifting, FlowMatch Euler scheduling), unified dispatch with Wan
- Support 4D image latents in VAE scheduler (skip temporal padding, ndim-based output dispatch)

## Accuracy Tests

```shell
python -u -m sgl_jax.launch_server \
  --multimodal \
  --model-path=/models/black-forest-labs/FLUX.1-dev

curl http://localhost:30000/api/v1/images/generation \
  -H "Content-Type: application/json" \
  -d '{"prompt": "A cat holding a sign that says hello world", "size": "832*480"}'
```

helloworldcat

### CODE REVIEW

File: python/sgl_jax/srt/multimodal/model_executor/diffusion/diffusion_model_runner.py [hunk 1]
```diff
@@ -87,74 +109,197 @@ def forward_model(
...
-            True if the request was aborted, False otherwise.
-        """
+        if self.is_flux():
+            return self._forward_flux(batch, mesh, abort_checker, step_callback)
+        else:
+            return self._forward_wan(batch, mesh, abort_checker, step_callback)
+
+    # ── FLUX forward ──
+
+    def _forward_flux(
```
Comment (pengchengneo) [inline python/sgl_jax/srt/multimodal/model_executor/diffusion/diffusion_model_runner.py:172]: there is some redundant code between _forward_flux and _forward_wan

we should seperate model specific implement and common code in different level

I will tiny refator it after this pr merged

**Review by pengchengneo — DISMISSED** [review]

File: python/sgl_jax/srt/multimodal/layers/rotary_embedding.py [hunk 2]
```diff
@@ -93,6 +93,42 @@ def __init__(
...
+        Calculates n-d rotary embeddings for explicit token coordinates.
+
+        Args:
+            pos: Tensor of shape [num_tokens, ndim] containing the coordinate of
+                each token on every rotary axis.
+
+        Returns:
+            A tuple of (cos, sin) tensors compatible with FLUX rotary apply.
+        """
+        pos = jnp.asarray(pos)
```
Comment (pengchengneo) [inline python/sgl_jax/srt/multimodal/layers/rotary_embedding.py]: why need this jnp.asarray?

File: python/sgl_jax/srt/multimodal/layers/visual_embedding.py [hunk 3]
```diff
@@ -172,3 +175,373 @@ def __call__(self, x: jax.Array) -> jax.Array:
...
+            mesh=mesh,
+            params_dtype=params_dtype,
+            kernel_axes=("tensor", None),
+        )
+        self.act_fn = act_fn
+        if act_fn == "gelu_tanh":
+            self.act_1 = modeling_flax_utils.ACT2FN["gelu_pytorch_tanh"]
+        elif act_fn == "silu":
+            self.act_1 = modeling_flax_utils.ACT2FN["silu"]
+        elif act_fn == "silu_fp32":
```
Comment (pengchengneo) [inline python/sgl_jax/srt/multimodal/layers/visual_embedding.py]: dead code?

File: python/sgl_jax/srt/multimodal/layers/rotary_embedding.py (hunk 2 above)
Comment (bjmsong) [reply python/sgl_jax/srt/multimodal/layers/rotary_embedding.py]: removed

File: python/sgl_jax/srt/multimodal/layers/visual_embedding.py (hunk 3 above)
Comment (bjmsong) [reply python/sgl_jax/srt/multimodal/layers/visual_embedding.py]: removed

**Review by pengchengneo — DISMISSED** [review]

**Review by pengchengneo — APPROVED** [review]
